### ISSUE TITLE
rdpproxy creates new folders on the client's drive, even if rdpdr_drive_write channel is denied.

### ISSUE THREAD

**Issue by kamelusz** [wallix/redemption#178]
Hello.

I compiled rdpproxy source code from commit **6de34f9e6047f5dd8d984804932c99022d8910b9** (**Nov 13, 2024**).
Created a configuration file (rdpproxy.ini) with the following contents:

```ini
[mod_rdp]
denied_channels = rdpdr_drive_write
allow_nla_ntlm_fallback = true
allow_tls_only_fallback = true

[server_cert]
server_cert_check = 3
```

After establishing the connection, I tried to perform some write operations on the client's drive. Almost all write operations were forbidden, except for creating a new folder through the context menu. See the video below.

<details>
On the left is explorer with the local drive on the computer where the rdp client is running. On the right, explorer on rdp server with the drive is forwarded to the one.

https://github.com/user-attachments/assets/a8ed770b-7c1a-4dcb-9cb9-5a8a2991e7c2
</details>

Server: Microsoft Windows 11 Pro, 10.0.22631 Build 22631.
Clients:
- mtsc (10.0.22621.4455);
- remmina (1.4.35).

**Comment by kamelusz** [comment]
Noted that new folder is created after processing such server message

```
rdpproxy: INFO (40056/40056) -- FileSystemVirtualChannel::process_server_message: total_length=80 flags=0x00000003 chunk_data_length=80
rdpproxy: INFO (40056/40056) -- FileSystemVirtualChannel::process_server_message: Server Drive I/O Request
rdpproxy: INFO (40056/40056) -- DeviceIORequest: DeviceId=3 FileId=0 CompletionId=4 MajorFunction=IRP_MJ_CREATE(0x0) MinorFunction=<unknown>(0x0)
rdpproxy: INFO (40056/40056) -- FileSystemVirtualChannel::process_server_drive_io_request: Device Create Request
rdpproxy: INFO (40056/40056) -- DeviceCreateRequest: DesiredAccess=0x100001 AllocationSize=0 FileAttributes=0x80 SharedAccess=0x3 CreateDisposition=0x2 CreateOptions=0x200021 Path="/New folder"
```

Added extra check to reject these messages.
```cpp
static inline
bool create_folder_is_required(uint32_t CreateOptions, uint32_t CreateDisposition) {
    if (!(CreateOptions & FILE_DIRECTORY_FILE)) {
        return false;
    }

    return CreateDisposition == FILE_CREATE;
}
```

It looks like this solves the problem with mtsc and freerdp based clients.

**Comment by XiaopengZHOU** [comment]
Hi kamelusz,

Thank you for your contribution.

I think checking the **CreateDisposition** flag is enough to resolve this issue.

**Comment by jonathanpoelen** [comment]
fixed in 4dcfc5a6def596d787e5937c54bda63a133f033a
